manifests/job.pp

```diff
@@ -25,12 +25,13 @@
 #     choose 'absent' to ensure the job is removed
 #
 define jenkins::job(
-  $config,
+  $config   = undef,
```

Comment:
I truly don't understand this approach, what we need here (in this defined type) is to have either `$config` or `$source` or `$template` to be defined (mandatory) and it can be tested (as per my proposal) and make the compilation fail if not respected. What we got here is a constraint on a required parameter which is not used in case we pass either `$source` or `$template`, so why has it to be kept as required ?